### PR TITLE
Architecture changed

### PR DESCRIPTION
**Pull request by Rodimkov**
Added a count of some metrics I'm not sure that this will be needed in the future. Also added a plot of the dependence of the error on the object and confusion matrix.

### CODE REVIEW

File: classification.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+            raise SystemExit(1)
+
+    def validate(self):
+        try:
+            if not 'processing_info' in self.data:
+                raise Exception('processing_info')
+            if not 'dataset_meta' in self.data:
+                raise Exception('dataset_meta')
+            if not 'report' in self.data:
+                raise Exception('report')
```
Comment (eaidova) [inline classification.py:51]: I believe, it is common structure for all reports, so it can be moved to base class

File: detection.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import cv2
+
+
+def draw_boxes(image, boxes, label_class, color=(255, 255, 255), thickness=2):
```
Comment (eaidova) [inline detection.py:4]: @Rodimkov, you do not update this file to use class or something went wrong during rebasing?

File: main.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                                 help="directory containing data")
+
+    return argument_parser
+
+
+def file_path(path):
+    if not os.path.exists(path):
+        raise argparse.ArgumentTypeError("{} is not a valid file".format(path))
+
+    _, file_extension = os.path.splitext(path)
```
Comment (eaidova) [inline main.py:23]: you can do check inisde argparsing

File: main.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+def main():
+    parser = create_parser()
+    namespace = parser.parse_args()
+
+    task = None
+
+    if namespace.temp == '0':
+        factory = task_factory.MetricAnalysisFactory()
+        task = factory.create_task("Classification", namespace.file[0], namespace.directory)
```
Comment (eaidova) [inline main.py:45]: since it is staticmethod, you no need to create object for using it
```
task = MetricAnalisysFacory.create_task(...)
```

File: classification.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+            self.identifier.append(report.get("identifier"))
+            self.prediction_label.append(report.get("prediction_label"))
+            self.annotation_label.append(report.get("annotation_label"))
+            self.prediction_scores.append(report.get("prediction_scores"))
+            self.accuracy_result.append(report.get("accuracy_result"))
+
+    def simple_metric(self):
+        from sklearn.metrics import accuracy_score, recall_score, precision_score, f1_score
+        
+        print(accuracy_score(self.prediction_label, self.annotation_label))
```
Comment (eaidova) [inline classification.py:89]: I think it can be useful to add the name of metric which you calculate in printing

File: detection.py (hunk 2 above)
Comment (Rodimkov) [reply detection.py:4]: The implementation is not yet ready for the detection task. It will look the same as for the classification task in the future.

File: classification.py [hunk 6]
```diff
@@ -1,25 +1,155 @@
...
+
+        for report in self.reports:
+            self.identifier.append(report.get("identifier"))
+            self.prediction_label.append(report.get("prediction_label"))
+            self.annotation_label.append(report.get("annotation_label"))
+            self.prediction_scores.append(report.get("prediction_scores"))
+            self.accuracy_result.append(report.get("accuracy_result"))
+
+    def top_n(self):
+        n = 10
```
Comment (eaidova) [inline classification.py:55]: on my view, `n` better to provide as parameter for method (you can make default value 10)

File: detection.py [hunk 7]
```diff
@@ -1,54 +1,252 @@
...
+            self.per_class_result.append(report.get("per_class_result"))
+
+        for info_class in self.per_class_result:
+            pb = {}
+            ab = {}
+            ps = {}
+            ap = {}
+            value = []
+
+            for key in info_class.keys():
```
Comment (eaidova) [inline detection.py:70]:
```suggestion
            for key, value in info_class.items():
```
